go/plugins/vertexai/modelgarden/modelgarden.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package modelgarden
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/firebase/genkit/go/genkit"
+)
+
+type ModelGardenOptions struct {
+	ProjectID string
+	Region    string
+	Models    []string
+}
+
+// Init initializes the ModelGarden plugin
+// After calling Init, you may call [DefineModel] to create and register
+// any additional generative models
+func Init(ctx context.Context, g *genkit.Genkit, cfg *ModelGardenOptions) error {
+	clients := NewClientFactory()
```

Comment:
note: maybe a list of clients should be returned to the caller so the instances are ready to be used later...